Nos/Service/Relay/Filter.swift

```diff
@@ -14,7 +14,10 @@ struct Filter: Hashable, Identifiable {
     var limit: Int?
     var since: Date?
     var until: Date?
-    
+
+    /// This filter should keep the subscription open.
```

Comment:
Since this isn't a simple on/off thing, I wonder if it would be easier to understand if we have a two-case enum. Maybe something like `FilterType` with cases named something like `subscribeToAll` and `oneTime` (or `oneAndDone`? 😂 Just kidding but I love the rhyme). I also understand that this may be overkill and the Bool might be fine with the right documentation like Matt suggested.

Reply:
In thinking more about this, I'm leaning more towards a `FilterType` enum (or should it be `SubscriptionType`?), maybe with cases named `allNewEvents` and `allStoredEvents` or just `newEvents` and `storedEvents`. I'm probably not the best to name this since I don't fully understand it, but that's how Matt's comment reads to me.

Reply:
The cases would be more like `storedEvents` and `storedEventsAndNewEvents`. I'm fine with the bool because these aren't really two disjoint sets of events. Both cases will return stored events and the question is whether we want to keep the subscription open and stream in new events - which is a boolean question. But I'm also not opposed to an enum if you feel it's more expressive.

Reply:
Is part of the problem that "subscribe" is too generic of a verb? Maybe `streamNewEvents` or `observeNewEvents` would be better?

Reply:
I'm not 100% sure, but yeah, I think it's too generic for me. Aren't we always subscribing, technically? And this `Bool` just determines whether we keep the subscription open? So maybe something along those lines, or maybe `observeNewEvents`.

Reply:
I'm changing it to `observeNewEvents` in RelaySubscription and `shouldKeepSubscriptionOpen` in Filter.